### PR TITLE
feat:migrate equipment deletion

### PR DESCRIPTION
**Pull request by laurgarn**

### CODE REVIEW

File: src/components/refactor/dialogs/equipment-deletion/equipment-deletion-dialog.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    const methods = useForm({
+        defaultValues: emptyFormData,
+        resolver: yupResolver(schema),
+    });
+
+    const { reset } = methods;
+
+    const fromEditDataToFormValues = useCallback(
+        (editData) => {
+            console.debug('EquipmentDeletion.fromEditDataToForm', editData);
```
Comment (ayolab) [inline src/components/refactor/dialogs/equipment-deletion/equipment-deletion-dialog.js:59]: remove this

**Review by ayolab — CHANGES_REQUESTED** [review]

File: src/components/refactor/dialogs/equipment-deletion/equipment-deletion-form.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        );
+        return ret;
+    }, []);
+
+    const watchType = useWatch({
+        name: TYPE,
+    });
+
+    const { setValue, getValues } = useFormContext();
+    useEffect(() => {
```
[the remark this answers is not in the document]
Comment (laurgarn) [reply src/components/refactor/dialogs/equipment-deletion/equipment-deletion-form.js]: Done.

File: src/components/refactor/dialogs/equipment-deletion/equipment-deletion-dialog.js (hunk 1 above)
Comment (laurgarn) [reply src/components/refactor/dialogs/equipment-deletion/equipment-deletion-dialog.js]: Done

File: src/components/refactor/dialogs/equipment-deletion/equipment-deletion-dialog.js [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+const schema = yup
+    .object()
+    .shape({
+        [TYPE]: yup.object().nullable().required(),
+        [EQUIPMENT_ID]: yup.string().nullable().required(),
+    })
+    .required();
+
+const emptyFormData = {
+    [TYPE]: null,
```
[the remark this answers is not in the document]
Comment (laurgarn) [reply src/components/refactor/dialogs/equipment-deletion/equipment-deletion-dialog.js]: Done

**Review by ayolab — APPROVED** [review]

**Review by archendel — APPROVED** [review]

**Pull request merged**